packages/@aws-cdk/cdk/lib/core/tokens.ts

```diff
@@ -29,6 +31,43 @@ export class Token {
 
         return value;
     }
+
+    /**
+     * Return a string representation of this token
+     *
+     * This string representation can be embedded into strings and
+     * will be turned into the actual representation of the token
+     * when resolve() is called on the string.
+     */
+    public toString(): string {
+        if (this.stringRepr === undefined) {
+            this.stringRepr = TOKEN_STRING_MAP.register(this);
+        }
+        return this.stringRepr;
+    }
+
+    /**
+     * Return the JSON representation of this Token
+     *
+     * Implements JSON.stringify()-compatibility for this Token. Note that it
+     * returns a string representation for the Token pre-resolution, NOT the
+     * resolved value of the Token.
+     */
+    public toJSON(): any {
+        return this.toString();
+    }
+}
+
+/**
+ * A Token that has a human-readable representation hint
+ *
+ * This can be used to give Tokens a string representation that makes
+ * sense to humans.
+ */
+export class HintedToken extends Token {
+    constructor(public readonly representationHint: string, valueOrFunction?: any) {
```

Comment:
Maybe just add an optional 2nd argument to the `Token` ctor and call it `displayName` or something like that, so this capability will be more discoverable by anyone defining a `Token`.

Would also simplify the whole duck-type-checks: `if (token.displayName) { bla }`